### PR TITLE
Re-export Element trait from selectors crate

### PR DESCRIPTION
**Pull request by adamreichold**
This is required to use its methods like `parent_element` from a dependent crate without adding a direct dependency to `selectors` to that crate.

### CODE REVIEW

File: src/node.rs [hunk 1]
```diff
@@ -29,7 +30,7 @@ pub enum Node {
     Text(Text),
 
     /// An element.
-    Element(Element),
+    Element(Box<Element>),
```
Comment (adamreichold) [inline src/node.rs:33]: This was done as a separate commit as the change to add heap allocations here might reasonably be considered controversial. The only alternative I see though, is to disable the lint for the `Node` enum.

Comment (cfvescovo) [reply src/node.rs:33]: I do not see any warning without `Box`ing `Element`. Could you paste here the warning you are getting? Thanks

Comment (adamreichold) [reply src/node.rs:33]: Yes, running the nightly toolchain, I get

```
warning: enum variant is more than three times larger (152 bytes) than the next largest
   --> src/node.rs:32:5
    |
32  |     Element(Element),
    |     ^^^^^^^^^^^^^^^^
    |
note: the lint level is defined here
   --> src/lib.rs:135:5
    |
135 |     variant_size_differences
    |     ^^^^^^^^^^^^^^^^^^^^^^^^
```

on the current `master` branch.

This seems reasonable as `Element` is more complicated than the other variants. I think the question whether `Element` is also much more common than the other variants so that boxing it will not really help with anything. (It only really helps if the boxed variant is so uncommon, that the cost of boxing it is offset by the speed-up when handling the other smaller variants.)

Comment (cfvescovo) [reply src/node.rs:33]: Now I can see it, thanks. As you said I do not think `Box`ing `Element` is going to improve performance as it is the most common variant. Heap allocating in this case provides a little benefit in terms of space at a higher time cost. I have not performed any benchmark so take what I am saying with a grain of salt

Comment (adamreichold) [reply src/node.rs:33]: I agree with that reasoning and will replace the commit by one just suppressing the lint.
